Quick_Deploy/vLLM/model_repository/vllm/1/model.py

```diff
@@ -0,0 +1,330 @@
+            top_p=request.top_p,
+            top_k=request.top_k,
+            stop=request.stop,
+            ignore_eos=request.ignore_eos,
+            max_tokens=request.max_tokens,
+        )
+
+    def make_output_options(self, req: CompletionRequest) -> OutputOptions:
+        """
+        Creates an OutputOptions object from the given arguments.
+        """
+        return OutputOptions(
+            echo=req.echo,
+            stream=req.stream,
+        )
+
+    def preprocess(self, request) -> Tuple[str, SamplingParams, OutputOptions]:
+        """
+        Converts a Triton request into a VLLM input by extracting the prompt and sampling parameters.
+        """
+        # deserialize the VLLM request from the Triton InferRequest
+        try:
+            serialized_json = pb_utils.get_input_tensor_by_name(
+                request, "serialized_request_json"
+            ).as_numpy()[0]
+            request = CompletionRequest(**json.loads(serialized_json))
+
+            return (
+                request.prompt,
+                self.make_sample_params(request),
+                self.make_output_options(request),
+            )
+        except Exception as e:
+            self.logger.log_info(f"Error parsing request: {e}")
+            raise e
+
+    def postprocess(
+        self, output: VLLMOutput, output_options: OutputOptions, inflight_avg: float
+    ):
+        """
+        Converts a VLLM output into a Triton response.
+        """
+        # create a Triton Response from the vLLM output
+        response = CompletionResponse()
+        response.request_id = output.request_id
+        response.finished = output.finished
+
+        # number of tokens in the prompt and total number of tokens generated across all completions
+        response.prompt_token_count = len(output.prompt_token_ids)
+
+        # optionally echo the prompt back to the client
+        if output_options.echo:
+            response.prompt = output.prompt
+
+        for completion in output.outputs:
+            completion_output = Completion(
+                index=completion.index,
+                text=completion.text,
+                finish_reason=completion.finish_reason,
+                cumulative_logprob=completion.cumulative_logprob,
+                gen_token_count=len(completion.token_ids),
+            )
+            response.completions.append(completion_output)
+
+        response.current_inflight_count = self._inflight_counter
+        response.average_inflight_count = inflight_avg
+
+        # serialize the NemoResponse into a tensor packed into Triton InferenceResponse
+        serialized_response = response.json().encode("utf-8")
+        triton_output_tensor = pb_utils.Tensor(
+            "serialized_response_json", np.array([serialized_response], dtype=np.object)
+        )
+
+        return pb_utils.InferenceResponse(output_tensors=[triton_output_tensor])
+
+    async def generate_stream(self, request):
```

Comment:
Question: Does it make sense to rename this to generate as it is used both for streaming as well as single response?

Reply:
I think `llm_inference` can be more comprehensible.

Reply:
good with either - I think generate maybe slightly more aligned with the llm_engine - 'inference' may be confused with a forward pass -